docs/integrations/aws-s3/locations.md

```diff
@@ -0,0 +1,28 @@
+---
+id: locations
+sidebar_label: Locations
+title: Amazon Web Services S3 Locations
+# prettier-ignore
+description: Setting up an integration with Amazon Web Services S3
+---
+
+The AWS S3 integration supports loading catalog entities from an S3 Bucket.
+Entities can be added to
+[static catalog configuration](../../features/software-catalog/configuration.md),
+or registered with the
+[catalog-import](https://github.com/backstage/backstage/tree/master/plugins/catalog-import)
+plugin.
+
+## Configuration
+
+To use this integration, add configuration to your root `app-config.yaml`:
+
+```yaml
+integrations:
+  awsS3:
+    accessKeyID: ${AWS_ACCESS_KEY_ID}
```

Comment:
It feels like this should allow for a list of host / credential configs rather then a single one. This would make it possible to use different credentials for per host as well as be more consistent with existing integrations.

Reply:
Hmmm agreed, but i wonder what config patterns are best to use then.

https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-bucket-intro.html

There are several possible url patterns that kinda overlap.

Will we need host + optional bucket as config params? Or making even host optional, implying the hosted amazonaws one in general if left out?

Reply:
That's true. There are a few more considerations then other URL readers in this regard. Using a public http url would likely require the bucket name as per this. It might also be nice to support the s3 protocol form as well at some point (e.g. `s3://some-bucket`). In this case, the bucket alone would need to be the indicator for which credentials to use.

Reply:
Yeah.

Maybe a good start is to check what parameters (including naming) that the node library accepts and consider mimicking

Reply:
Constructing the URL dynamically with strict validation would exclude many AWS-S3-like services from the mix as well. These alternatives use the same API as S3 so a small modification here would enable those as well with this same PR (with the addition of forcePathStyle config option). I'd recommend using a similar structure as in `plugins/techdocs-backend/config.d.ts`

Reply:
Ahh of course, there's prior art - thanks for reminding me, yes let's standardize how we express this.

Reply:
We made the AWS integration support a list of host / credential configs. To do this, we followed what had been done for the other URL readers that support a list like adding the `AwsS3Integrations` class that implements `ScmIntegration` . The other Url readers that are included in `ScmIntegrations` seem to be associated with git repositories while AWS S3 is not. Is this a proper use of `ScmIntegrations`?

Reply:
I think you are doing the right thing here. This will open it up for usage in the URL readers automatically etc. I think the SCM part may have been a bit too strict naming on our part.